test/li_test.jl

```diff
@@ -0,0 +1,144 @@
+using SpecialFunctions
+using Base.Test
+
+const SF = SpecialFunctions
+
+
+# golden ratio is used in some tests
+ϕ = (sqrt(5)+1)/2.0
+
+@testset "Polylogarithm Li" begin
+    @testset "Check different input formats" begin
+        @test SF.Li(-1.0, 0.0) ≈ 0.0
+        @test SF.Li(-1, 0.0) ≈ 0.0
+        @test SF.Li(-1.0, Complex(0.0)) ≈ 0.0 + 0.0im # don't seem to need imaginary bit here
+        @test SF.Li(Complex(-1.0), 0.0) ≈ 0.0 + 0.0im
+        @test SF.Li(Complex(-1.0), 0.0) ≈ 0.0
+        @test SF.Li(1, 0) ≈ 0.0
+        x = collect(0.0:0.1:0.9)
+        @test all([SF.Li.(1, x)[i] ≈ SF.Li(1, x[i]) for i=1:length(x)])
+        @test SF.Li(Complex(-1.0), Complex(0.3)) ≈ SF.Li(-1.0, 0.3)
```

Comment:
`≈` is not sufficient for checking a special-function implementation; it only checks that about half of the digits are correct.  Use `≅`, defined in runtests.jl, instead: it checks that both the real and imaginary parts have about 13 digits.

And it is not sufficient to check symmetries and simple values like this.  You need to check points just on either side of the thresholds between the different approximations that you use, points close to the real or imaginary axes (e.g. `1 + 1e-15*im`) etcetera.  Use e.g. WolframAlpha to get accurate to test against.

Another good thing to try would be to run e.g. 10^5 random points in the complex plane against some known implementation (e.g. the one in Python's `mpmath`) and check the maximum `relerrc` (defined in runtests.jl).